### PR TITLE
feat: add radar history slider

### PR DESCRIPTION
**Pull request by MaxProtos**

### CODE REVIEW

File: src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+@Component({
+	selector: 'app-radar-date-switcher',
+	templateUrl: './radar-date-switcher.component.html',
+	styleUrls: ['./radar-date-switcher.component.scss'],
+})
+export class RadarDateSwitcherComponent implements OnInit, OnDestroy {
+	public theme$: Observable<ComponentTheme>;
+	public radarDate: Date;
+
+	private destroy$: Subject<boolean> = new Subject<boolean>();
```
Comment (clain-atomi) [inline src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.ts:19]: Subject`<void>`

File: src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+	public switchNextRadarDay(): void {
+		const newRadarDate: Date = new Date(this.radarDate);
+		newRadarDate.setDate(this.radarDate.getDate() + this.consideredNewInDays);
+
+		this.radarViewFacadeService.updateRadarDate(newRadarDate);
+	}
+
+	public ngOnDestroy(): void {
+		this.destroy$.next(true);
```
Comment (clain-atomi) [inline src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.ts:70]: .next()

File: src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+		this.destroy$.unsubscribe();
+	}
+
+	private updateRouteQueryParams(date: Date): void {
```
Comment (clain-atomi) [inline src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.ts:74]: updateDateInUrl

File: src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		const currentDate: Date = new Date();
+		const dateParam: string = `${date.getUTCFullYear()}-${
```
Comment (clain-atomi) [inline src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.ts:76]: date param = date.toUtcString()

File: src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+			date.getUTCMonth() + 1
+		}-${date.getUTCDate()}-${currentDate.getUTCHours()}-${currentDate.getUTCMinutes()}-GMT`;
+		this.router.navigate([], {
+			relativeTo: this.activatedRoute,
+			queryParams: { date: dateParam },
+			queryParamsHandling: 'merge',
+		});
+	}
+
+	private updateRadarDateFromUrlParams(date: string, radarId: string): void {
```
Comment (clain-atomi) [inline src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.ts:86]: updateRadarDateFromUrl

File: src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+		const dateAsArray: string[] = date.split('-');
+		const radarDateFromParams: Date = new Date();
+		const yearFromParams: number = parseInt(dateAsArray[0], 10);
+		const monthFromParams: number = parseInt(dateAsArray[1], 10) - 1;
+		const dateFromParams: number = parseInt(dateAsArray[2], 10);
```
Comment (clain-atomi) [inline src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.ts:91]: parsedDate: Date = new Date(date);

File: src-fe/src/app/libs/radar-view/components/radar-chart/radar-chart.component.html [hunk 7]
```diff
@@ -6,11 +6,13 @@
 						[theme]="containerFacade.theme$ | async">
 	</app-dot-tooltip>
 </app-tooltip>
-	
 
-<app-zoom-in-out-panel	class="radar-chart__zoom-in-out" 
+
+<app-zoom-in-out-panel	class="radar-chart__zoom-in-out"
```
Comment (clain-atomi) [inline src-fe/src/app/libs/radar-view/components/radar-chart/radar-chart.component.html:11]: align indentation

File: src-fe/src/app/libs/radar-view/components/radar-chart/radar-chart.component.scss [hunk 8]
```diff
@@ -11,4 +11,11 @@
 		bottom: $gutter-width;
 		right: $gutter-width;
 	}
+
+	&__date-switcher {
+		position: absolute;
+		bottom: $gutter-width;
+		right: calc(#{$gutter-width} + 70px);
```
Comment (clain-atomi) [inline src-fe/src/app/libs/radar-view/components/radar-chart/radar-chart.component.scss:18]: 70px = $secondary-grid-unit * ...

File: src-fe/src/app/libs/radar-view/store/radar-view/radar-view.actions.ts [hunk 9]
```diff
@@ -32,7 +34,7 @@ export class ShowUploadErrors implements Action {
 
 export class LoadRadar implements Action {
 	public readonly type: RadarViewActionTypes = RadarViewActionTypes.LoadRadar;
-	constructor(public payload: string) {}
+	constructor(public radarId: string, public date?: Date) {}
```
Comment (clain-atomi) [inline src-fe/src/app/libs/radar-view/store/radar-view/radar-view.actions.ts:37]: merge to single param

Comment (clain-atomi) [reply src-fe/src/app/libs/radar-view/store/radar-view/radar-view.actions.ts:37]: date is not optional

File: src-fe/src/app/libs/radar-view/store/radar-view/radar-view.effects.ts [hunk 10]
```diff
@@ -33,15 +33,26 @@ export class RadarViewEffects {
 	@Effect()
 	public loadRadar$: Observable<Action> = this.actions$.pipe(
 		ofType(RadarViewActionTypes.LoadRadar),
-		switchMap((action: LoadRadar) => {
-			return this.radarsRepositoryService.loadRadar(action.payload, new Date()).pipe(
+		withLatestFrom(this.radarViewFacadeService.radarDate$),
+		switchMap(([action, radarDate]: [LoadRadar, Date]) => {
+			const dateToLoad: Date = action.date ?? radarDate;
```
Comment (clain-atomi) [inline src-fe/src/app/libs/radar-view/store/radar-view/radar-view.effects.ts:38]: selectedDate

File: src-fe/src/app/libs/radar-view/store/radar-view/radar-view.effects.ts [hunk 11]
```diff
@@ -33,15 +33,26 @@ export class RadarViewEffects {
...
 					return new LoadRadarSuccess(this.radarConverterService.fromDto(dto));
 				})
 			);
 		})
 	);
 
+	@Effect()
+	public changeRadarDate$: Observable<Action> = this.actions$.pipe(
+		ofType(RadarViewActionTypes.ChangeRadarDate),
+		withLatestFrom(this.radarViewFacadeService.radar$),
```
Comment (clain-atomi) [inline src-fe/src/app/libs/radar-view/store/radar-view/radar-view.effects.ts:50]: add filter pipe to check that radar is exists

File: src-fe/src/app/libs/radar-view/store/radar-view/radar-view.reducer.ts [hunk 12]
```diff
@@ -18,17 +20,20 @@ export const initialState: RadarViewState = {
 	searchQuery: null,
 	filteredRadarDataItems: null,
 	uploadErrors: null,
+	radarDate: new Date(),
 };
 
 export function radarViewReducer(state: RadarViewState = initialState, action: RadarViewActions): RadarViewState {
 	switch (action.type) {
 		case RadarViewActionTypes.LoadRadar: {
+			const dateFromQueryParams: Date = (action as LoadRadar).date ?? null;
```
Comment (clain-atomi) [inline src-fe/src/app/libs/radar-view/store/radar-view/radar-view.reducer.ts:29]: selectedDate

**Review by clain-atomi — CHANGES_REQUESTED** [review]

File: src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.html [hunk 13]
```diff
@@ -0,0 +1,18 @@
+<div class="radar-date-switcher" [class.radar-date-switcher_dark-theme]="(theme$ | async) === 'dark'">
+	<div class="radar-date-switcher__date-container">
+		<div class="radar-date-switcher__date-toggle radar-date-switcher__date-toggle_left"
+			 (click)="switchPreviousRadarDay()">
+			<app-icon class="radar-date-switcher__arrow-icon"
+					  icon="arrow-left3">
```
[the remark this answers is not in the document]
Comment (MaxProtos) [reply src-fe/src/app/libs/common-components/radar-date-switcher/radar-date-switcher.component.html:6]: github issues

File: src-fe/src/app/libs/radar-view/components/date-switcher/date-switcher.component.html [hunk 14]
```diff
@@ -0,0 +1,18 @@
+<div class="date-switcher" [class.date-switcher--dark-theme]="(theme$ | async) === 'dark'">
```
Comment (clain-atomi) [inline src-fe/src/app/libs/radar-view/components/date-switcher/date-switcher.component.html:1]: move to getter `(theme$ | async) === 'dark'"` also better use `theme` as @Input to allow change theme dynamically

**Review by clain-atomi — CHANGES_REQUESTED** [review]

**Review by clain-atomi — APPROVED** [review]

**Pull request merged**